### ISSUE TITLE
flush isn't exposed for TransactionDB

### ISSUE THREAD

**Issue by kayabaNerve** [rust-rocksdb/rust-rocksdb#876]
It's solely exposed for DBCommon, backing DBWithThreadMode and OptimisticTransactionDB (yet not TransactionDB).

The raw FFI call exists in librocksdb-sys, yet there's no safe way to get the pointer for the TransactionDB from the object.

This, AFAICT, prevents ever flushing TransactionDBs to disk and causes massive WALs (due to them never being flattened into an SST).

**Comment by pragmaxim** [comment]
@kayabaNerve I agree, the data loss is massive :cry: Did you figure out the status of it? Or we simply cannot use TransactionDB now ?

**Comment by kayabaNerve** [comment]
I believe I moved to OptimisticTransactionDB as my use was 'safe' (I don't write to the same keys across distinct TXs, and I'm fine panicking if it turns out I do).

**Comment by kayabaNerve** [comment]
https://doc.rust-lang.org/std/ops/trait.Drop.html

**Comment by pragmaxim** [comment]
@kayabaNerve the `db.flush` does not even work in OptimisticTransactionDB : https://github.com/rust-rocksdb/rust-rocksdb/issues/900

**Comment by kayabaNerve** [comment]
*does not work with the OTDB when you don't have a WAL, according to that issue. Have you observed this for an OTDB regardless of the WAL?

**Comment by pragmaxim** [comment]
Yes, just tried in the rust-playground, it happens regardless of WAL being on/off ... the `db.flush` call simply does not lead to new sst file in the db directory, therefore nothing is recovered if you stop/start the process

**Comment by kayabaNerve** [comment]
Hm. I believe I observed SSTs in my work.

https://github.com/serai-dex/serai/blob/develop/common/db/src/rocks.rs#L30-L31
https://github.com/serai-dex/serai/blob/develop/common/db/src/rocks.rs#L50-L66

for my exact config.

**Comment by pragmaxim** [comment]
@kayabaNerve hm, I am using cutting edge revision, I have to give up for now : 
```
rocksdb = { git = "https://github.com/rust-rocksdb/rust-rocksdb", default-features = false, features = [
    "multi-threaded-cf",
], rev = "6943a2d06064780a26c6ec581c3182bc8e2f6a5a" }
```

**Comment by rayunstop** [comment]
hi @kayabaNerve , do u add the api for transactionDB already? how's the issue now
